internal/isoeditor/rhcos.go

```diff
@@ -130,6 +146,29 @@ func (e *rhcosEditor) addCustomRAMDisk(staticIPConfig string) error {
 	return editFile(e.isoHandler.ExtractedPath("isolinux/isolinux.cfg"), `(?m)^(\s+append.*initrd=\S+) (.*)$`, fmt.Sprintf("${1},%s ${2}", imagePath))
 }
 
+func (e *rhcosEditor) formatCurlConfigFile(clusterProxyInfo *isoutil.ClusterProxyInfo) (string, error) {
+	var httpProxy string
+	if clusterProxyInfo.HTTPSProxy != "" {
+		httpProxy = clusterProxyInfo.HTTPSProxy
+	} else if clusterProxyInfo.HTTPProxy != "" {
+		httpProxy = clusterProxyInfo.HTTPProxy
+	}
```

Comment:
Setting HTTP proxy to the value of HTTPS proxy is a bug. It works the other way around - if HTTPS proxy was not set, then HTTP proxy will be used for HTTPS traffic. Most tools do this without a need to set HTTPS proxy explicitly. I.e. is a HTTPS proxy was specified, use it for HTTPS, otherwise use HTTP proxy. See, for example, #3.2 of https://docs.openshift.com/container-platform/4.6/operators/admin/olm-configuring-proxy-support.html

Reply:
Here's how it's done in Golang https://github.com/golang/net/blob/release-branch.go1.15/http/httpproxy/proxy.go#L130

Reply:
curl supports only a single 'proxy' property in curlrc config file, which takes either https or http protocol.
So I'm setting the 'proxy' to either https or http, with a priority to https. 
This is just for downloading the rootfs image (boot-files API), so we can just use HTTPS for it.